lib/node_modules/@stdlib/blas/ext/base/dapxsumkbn-wasm/lib/routine.js

```diff
@@ -0,0 +1,168 @@
+*
+* // Create a new routine:
+* var dapxsumkbn = new Routine();
+*
+* // Initialize the module:
+* dapxsumkbn.initializeSync();
+*
+* // Define a strided array:
+* var x = new Float64Array( [ 2.0, 1.0, 2.0, -2.0, -2.0, 2.0, 3.0, 4.0 ] );
+*
+* // Perform operation:
+* var sum = dapxsumkbn.ndarray( 4, 5.0, x, 2, 1 );
+* // returns 25.0
+*/
+function Routine() {
+	if ( !( this instanceof Routine ) ) {
+		return new Routine();
+	}
+	Module.call( this, new Memory({
+		'initial': 0
+	}));
+	return this;
+}
+
+// Inherit from the parent constructor:
+inherits( Routine, Module );
+
+/**
+* Adds a scalar constant to each double-precision floating-point strided array element and computes the sum using an improved Kahan–Babuška algorithm.
+*
+* @name main
+* @memberof Routine.prototype
+* @readonly
+* @type {Function}
+* @param {PositiveInteger} N - number of indexed elements
+* @param {number} alpha - constant
+* @param {Float64Array} x - input array
+* @param {integer} strideX - stride length
+* @returns {number} sum
+*
+* @example
+* var Float64Array = require( '@stdlib/array/float64' );
+*
+* // Create a new routine:
+* var dapxsumkbn = new Routine();
+*
+* // Initialize the module:
+* dapxsumkbn.initializeSync();
+*
+* // Define a strided array:
+* var x = new Float64Array( [ 1.0, -2.0, 2.0 ] );
+*
+* // Perform operation:
+* var sum = dapxsumkbn.main( 3, 5.0, x, 1 );
+* // returns 16.0
+*/
+setReadOnly( Routine.prototype, 'main', function dapxsumkbn( N, alpha, x, strideX ) {
+	return this.ndarray( N, alpha, x, strideX, stride2offset( N, strideX ) );
+});
+
+/**
+* Adds a scalar constant to each double-precision floating-point strided array element and computes the sum using an improved Kahan–Babuška algorithm and alternative indexing semantics.
+*
+* @name ndarray
+* @memberof Routine.prototype
+* @readonly
+* @type {Function}
+* @param {PositiveInteger} N - number of indexed elements
+* @param {number} alpha - constant
+* @param {Float64Array} x - input array
+* @param {integer} strideX - stride length
+* @param {NonNegativeInteger} offsetX - starting index
+* @returns {number} sum
+*
+* @example
+* var Float64Array = require( '@stdlib/array/float64' );
+*
+* // Create a new routine:
+* var dapxsumkbn = new Routine();
+*
+* // Initialize the module:
+* dapxsumkbn.initializeSync();
+*
+* // Define a strided array:
+* var x = new Float64Array( [ 2.0, 1.0, 2.0, -2.0, -2.0, 2.0, 3.0, 4.0 ] );
+*
+* // Perform operation:
+* var sum = dapxsumkbn.ndarray( 4, 5.0, x, 2, 1 );
+* // returns 25.0
+*/
+setReadOnly( Routine.prototype, 'ndarray', function dapxsumkbn( N, alpha, x, strideX, offsetX ) {
+	var ptrs;
+	var p0;
+
+	// Convert the input arrays to "pointers" in the module's memory:
+	ptrs = arrays2ptrs( this, [
+		strided2object( N, x, strideX, offsetX )
+	]);
+	p0 = ptrs[ 0 ];
+
+	// Perform computation by calling the corresponding parent method:
+	return Module.prototype.ndarray.call( this, N, alpha, p0.ptr, p0.stride, p0.offset );
+});
```

Comment:
I can't exactly pinpoint what I'm doing wrong here, but tests are failing for cases when `stride` is zero. The compiled `wasm` functions are working as intended, as the corresponding `Module.prototype` tests are passing.

Reply:
@Snehil-Shah you should close this PR and reopen it , It can work

Reply:
@Snehil-Shah ok

Reply:
Hmm...based on CI, I am not sure that the wasm binary corresponds to the upstream C implementation. The test result is actually correct. Namely, when the stride is `0`, the result should actually be `N * ( alpha + x[ offsetX ] )`, not `alpha + x[ offsetX ]`. We've been changing that in the various extended BLAS packages, but it doesn't appear that `dapxsumkbn` has been updated, yet, on `develop`.

Reply:
@kgryte I am updating it, should be ready in a few mins.

Reply:
Regardless, it is still odd that the actual is the expected value, which makes me believe that the Wasm binary is of a different implementation. @Snehil-Shah Would you mind running `make clean-wasm` on the package and then recompiling?

Reply:
@kgryte just did, no change..